app/controllers/tournaments_controller.rb

```diff
@@ -28,10 +28,10 @@ def register
   def create_matches
     registered_users = @tournament.users
     length = registered_users.length
-    return unless (length - 8) != 0
+    return if (length - 8) != 0
```

Comment:
```suggestion
    return if length != 8
```

Check if length is power of 2 or not.